src/index.js

```diff
@@ -20,9 +20,10 @@ const {
 * @param  {string}   authURL   The URL to authenticate against
 * @param  {string}   username  The username to test
 * @param  {string}   password  The password to test
+* @param  {string}   machineUUID The machine UUID to check - optional
 * @return {object}             JSON formatted response object
 */
-export async function authenticateAsync(authURL, username, password) {
+export async function authenticateAsync(authURL, username, password, machineUUID = '') {
```

Comment:
@raviSussol -> can the last parameter be a little bit more dynamic, i suggest making it a JSON.

i.e.
```
export async function authenticateAsync(authURL, username, password, extraHeaders = {})
...
   Authorization: getAuthHeader(username, password),
   ...extraHeaders  
```